SU2_CFD/src/variables/CVariable.cpp

```diff
@@ -37,155 +37,73 @@
    to allocate some extra flow variables that do not participate
    in the simulation ---*/
-  Solution = new su2double [nVar];
+  Solution.resize(nPoint,nVar) = su2double(0.0);
 
-  for (iVar = 0; iVar < nVar; iVar++)
-    Solution[iVar] = 0.0;
+  Solution_Old.resize(nPoint,nVar);
 
-  Solution_Old = new su2double [nVar];
-
-  Gradient = new su2double* [nVar];
-  for (iVar = 0; iVar < nVar; iVar++) {
-    Gradient[iVar] = new su2double [nDim];
-    for (iDim = 0; iDim < nDim; iDim ++)
-      Gradient[iVar][iDim] = 0.0;
-  }
+  Gradient.resize(nPoint,nVar,nDim,0.0);
 
   if (config->GetUnsteady_Simulation() != NO) {
-    Solution_time_n = new su2double [nVar];
-    Solution_time_n1 = new su2double [nVar];
+    Solution_time_n.resize(nPoint,nVar);
+    Solution_time_n1.resize(nPoint,nVar);
   }
   else if (config->GetDynamic_Analysis() == DYNAMIC) {
-    Solution_time_n = new su2double [nVar];
-    for (iVar = 0; iVar < nVar; iVar++) Solution_time_n[iVar] = 0.0;
+    Solution_time_n.resize(nPoint,nVar) = su2double(0.0);
   }
 
-	if (config->GetFSI_Simulation() && config->GetDiscrete_Adjoint()){
-	  Solution_Adj_Old = new su2double [nVar];
+	if (config->GetFSI_Simulation() && config->GetDiscrete_Adjoint()) {
+	  Solution_Adj_Old.resize(nPoint,nVar);
 	}
 
   if (config->GetKind_Gradient_Method() == WEIGHTED_LEAST_SQUARES) {
-    Rmatrix = new su2double*[nDim];
-    for (iDim = 0; iDim < nDim; iDim++) {
-      Rmatrix[iDim] = new su2double[nDim];
-      for (jDim = 0; jDim < nDim; jDim++)
-        Rmatrix[iDim][jDim] = 0.0;
-    }
+    Rmatrix.resize(nPoint,nDim,nDim,0.0);
   }
 
+  Non_Physical.resize(nPoint) = false;
+
 }
 
-CVariable::~CVariable(void) {
-  unsigned short iVar, iDim;
-
-  if (Solution            != NULL) delete [] Solution;
-  if (Solution_Old        != NULL) delete [] Solution_Old;
-  if (Solution_time_n     != NULL) delete [] Solution_time_n;
-  if (Solution_time_n1    != NULL) delete [] Solution_time_n1;
-  if (Limiter             != NULL) delete [] Limiter;
-  if (Solution_Max        != NULL) delete [] Solution_Max;
-  if (Solution_Min        != NULL) delete [] Solution_Min;
-  if (Grad_AuxVar         != NULL) delete [] Grad_AuxVar;
-  if (Undivided_Laplacian != NULL) delete [] Undivided_Laplacian;
-  if (Res_TruncError      != NULL) delete [] Res_TruncError;
-  if (Residual_Old        != NULL) delete [] Residual_Old;
-  if (Residual_Sum        != NULL) delete [] Residual_Sum;
-  if (Solution_Adj_Old    != NULL) delete [] Solution_Adj_Old;
-
-  if (Gradient != NULL) {
-    for (iVar = 0; iVar < nVar; iVar++)
-      delete [] Gradient[iVar];
-    delete [] Gradient;
-  }
+void CVariable::Set_OldSolution() { Solution_Old = Solution; }
 
-  if (Rmatrix != NULL) {
-    for (iDim = 0; iDim < nDim; iDim++)
-      delete [] Rmatrix[iDim];
-    delete [] Rmatrix;
-  }
+void CVariable::Set_Solution() { Solution = Solution_Old; }
 
-}
+void CVariable::Set_Solution_time_n() { Solution_time_n = Solution; }
```

Comment:
Some methods that were used to operate on all nodes are now in the cpp and their use no longer requires writing a for loop.